backend/main.py

```
@@ -521,4 +603,193 @@ def buy_item(
+        raise HTTPException(status_code=400, detail=str(e)) from None
+    except Exception:
+        logger.exception("Failed to create quest")
+        raise HTTPException(status_code=500, detail="Failed to create quest") from None
+
+
+@app.post("/quests/{quest_id}/accept", response_model=schemas.QuestResponse)
+def accept_quest(
+    quest_id: UUID,
+    payload: schemas.QuestAcceptRequest,
+    current_user: Annotated[models.User, Depends(get_current_user_model)],
+    db: Session = Depends(get_db),
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)